contrib/parseq-batching/src/main/java/com/linkedin/parseq/batching/BatchImpl.java

```diff
@@ -0,0 +1,148 @@
+package com.linkedin.parseq.batching;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import com.linkedin.parseq.promise.PromiseResolvedException;
+import com.linkedin.parseq.promise.Promises;
+import com.linkedin.parseq.promise.SettablePromise;
+import com.linkedin.parseq.trace.ShallowTraceBuilder;
+
+public class BatchImpl<K, T> implements Batch<K, T> {
+
+  private final Map<K, BatchEntry<T>> _map;
+
+  private BatchImpl(Map<K, BatchEntry<T>> map) {
+    _map = map;
+  }
+
+  @Override
+  public void done(K key, T value) throws PromiseResolvedException {
+    _map.get(key).getPromise().done(value);
```

Comment:
yeah, my question is basically what if user calls done on a non-exist key so that _map.get(key) is null? shall we also provide a boolean contains(K key) method like the Map?